src/isa/exec.rs

```diff
@@ -469,14 +469,9 @@ impl InstructionSet for ArithmeticOp {
                 reg,
                 idx,
                 regs.get(reg, idx).and_then(|val| {
-                    if step.as_i16() < 0 {
-                        val.int_sub(Number::from(-step.as_i16()), IntFlags {
```

Comment:
When applying `Stp` to `A8` register, we are not sure that the step value (which is stored as `i16`) is within the scope of 8 bits. Maybe we should make `Step` `i8`, or add a method on `Number` that check if a number is representable in target bits (`reshape` method does not check that. it always returns false if a target bits are less than current bits)?

Reply:
Very good point! I think we need to reduce step down to u8 - but we need to make sure that the byte command is updated correspondingly. Let's do it as a separate PR (or pls open an issue for that just not to forget)